framework/src/problems/FEProblemBase.C

```diff
@@ -2955,6 +2972,22 @@ FEProblemBase::computeUserObjects(const ExecFlagType & type, const Moose::AuxGro
   // Finalize, threadJoin, and update PP values of Nodal
   finalizeUserObjects<NodalUserObject>(nodal);
 
+  // Initialize threaded GeneralUserObjects
+  initializeUserObjects<GeneralUserObject>(threaded_general);
+
+  if (threaded_general.hasActiveObjects())
+  {
+    std::vector<THREAD_ID> thread_ids(libMesh::n_threads());
+    for (THREAD_ID tid = 0; tid < libMesh::n_threads(); ++tid)
+      thread_ids[tid] = tid;
+
+    ComputeThreadedGeneralUserObjectsThread ctguot(*this, threaded_general);
+    Threads::parallel_reduce(ThreadIdRange(thread_ids.begin(), thread_ids.end()), ctguot);
```

Comment:
Ok - the new comment is that this doesn't work with interdependencies.  Each ThreadedGeneralUserObject needs its own thread loop - then call `threadjoin()` and `finalize()` on it before starting the next one.